sklearn/metrics/_classification.py

```diff
@@ -2013,63 +2038,110 @@ class after being classified as negative. This is the case when the
     >>> class_likelihood_ratios(y_true, y_pred, labels=["non-cat", "cat"])
     (1.5, 0.75)
     """
-
+    # TODO(1.8): When `raise_warning` is removed, the default return value with
+    # zero_division="warn" does not need to be "np.nan" anymore and should be the lowest
+    # score for each metric respectively (1 for LR+ and 0 for LR-) to match the other
+    # functions that take a zero_division param. Return values and warning messages need
+    # to be updated. The Warns section in the docstring needs be be re-written and
+    # the hidden option for zero_division ("default") be removed and the default set to
+    # "warn" to match the other functions that take a zero_division param.
     y_type, y_true, y_pred = _check_targets(y_true, y_pred)
     if y_type != "binary":
         raise ValueError(
             "class_likelihood_ratios only supports binary classification "
             f"problems, got targets of type: {y_type}"
         )
 
+    if raise_warning != "deprecated":
+        if zero_division != "default":
+            raise ValueError(
+                "`zero_division` and `raise_warning` cannot both be set. "
+                "Use either one to control the return values in case "
+                "of a division by zero."
+            )
+
+        warnings.warn(
+            "`raise_warning` was deprecated in version 1.6 and will be "
+            "removed in 1.8. To define the return values in case of a "
+            "division by zero, use `zero_division` instead.",
+            FutureWarning,
+        )
+
+        if raise_warning:  # user explicitly set `raise_warning=True`
+            zero_division = "warn"
+
+        else:  # user explicitly set `raise_warning=False`
+            zero_division = {"LR+": np.nan, "LR-": np.nan}
+
+    if zero_division == "default":
+        zero_division = "warn"
+
     cm = confusion_matrix(
         y_true,
         y_pred,
         sample_weight=sample_weight,
         labels=labels,
     )
 
-    # Case when `y_test` contains a single class and `y_test == y_pred`.
-    # This may happen when cross-validating imbalanced data and should
-    # not be interpreted as a perfect score.
-    if cm.shape == (1, 1):
-        msg = "samples of only one class were seen during testing "
-        if raise_warning:
-            warnings.warn(msg, UserWarning, stacklevel=2)
+    tn, fp, fn, tp = cm.ravel()
+    support_pos = tp + fn
+    support_neg = tn + fp
+    pos_num = tp * support_neg
+    pos_denom = fp * support_pos
+    neg_num = fn * support_neg
+    neg_denom = tn * support_pos
+
+    if support_pos == 0:
+        msg = (
+            "No samples of the positive class are present in `y_true`. "
+            "`positive_likelihood_ratio` and `negative_likelihood_ratio` are both set "
+            "to np.nan."
+        )
+        warnings.warn(msg, UserWarning, stacklevel=2)
```

Comment:
I think that we should use `UndefinedMetricWarning` to be consistent with other warning. I would probably consider that as a fix (we should add this change in the changelog).